src/pages/en/guides/typescript.md

```diff
@@ -78,11 +78,38 @@ export interface Props {
   name: string;
   greeting?: string;
 }
-const { greeting = 'Hello', name } = Astro.props as Props
+const { greeting = 'Hello', name } = Astro.props as Props;
 ---
 <h2>{greeting}, {name}!</h2>
 ```
 
+### Built-in attribute types
+
+Astro provides JSX type definitions to check your markup is using valid HTML attributes. You can use these types to help build component props. For example, if you were building a `<Link>` component, you could do the following to mirror the default HTML attributes in your component’s prop types.
+
+```astro
+---
+export type Props = astroHTML.JSX.HTMLAttributes<HTMLAnchorElement>;
+const { href, ...attrs } = Astro.props as Props;
+---
+<a {href} {...attrs}>
+  <slot />
+</a>
+```
+
+It is also possible to extend the default JSX definitions to add non-standard attributes by redeclaring the `astroHTML.JSX` namespace in a `.d.ts` file.
+
+```ts
+// src/custom-attributes.d.ts
+
+declare namespace astroHTML.JSX {
+  interface HTMLAttributes {
+    'data-count'?: number;
+    'data-label'?: string;
```

Comment:
This is fine to add type checking, but `data-` attributes don't need to be added to `HTMLAttributes` to avoid errors as attributes with a dash in them are always valid (similarly to how elements with a dash in them are always valid)

<img width="298" alt="image" src="https://user-images.githubusercontent.com/3019731/179211341-7245f0fb-07d8-4b45-b7c3-b3c6d31fbce5.png">

Reply:
Hmm, not necessarily. The specific use case I'm thinking about was [this user who needed a new global attribute that we didn't have yet](https://github.com/withastro/astro/pull/3801#issuecomment-1172889426). I'd think it's more common in general to add a new global attribute than an element specific one, not sure though

Reply:
TIL `inert`!

OK, I’ll leave this example as is. I know `data-` attributes were already supported, so this just adds types, but I thought that seemed like a vaguely useful example (maybe you built a client-side script that uses specific `data-` attributes or something) instead of a totally abstract one.

Thank you!